src/main/java/teammates/ui/controller/StudentCommentsPageAction.java

```diff
@@ -153,7 +158,40 @@ private void setNextPageLink(List<CourseAttributes> courses, int currentIndex){
                 }
             }
         }
-        return feedbackResultBundles;
+        return sortFeedbackBundlesByEndTimeReverse(feedbackResultBundles);
+    }
+    
+    /**
+     * @param feedbackResultBundles
+     *            The map of FeedbackSessionResultsBundle which is to be sorted
+     * @return Returns a sorted map(LinkedHashMap) of
+     *         FeedbackSessionResultsBundle, where the sessions are sorted in
+     *         descending order of their endTime
+     */
+    private Map<String, FeedbackSessionResultsBundle> sortFeedbackBundlesByEndTimeReverse(
+            Map<String, FeedbackSessionResultsBundle> feedbackResultBundles) {
+
+        List<Entry<String, FeedbackSessionResultsBundle>> list = new ArrayList<Entry<String, FeedbackSessionResultsBundle>>(
+                feedbackResultBundles.entrySet());
+        Collections.sort(list,
+                new Comparator<Entry<String, FeedbackSessionResultsBundle>>() {
```

Comment:
@kai33 Thanks for your quick review.

Yeah a comparator for FeedbackSessionResultsBundle can be used.

Also, I don't think a TreeMap can be used directly here. Since a TreeMap maintains an ordering on its keys. But here we are sorting by values of the inital Map.

Also, I think that the usage of Map< String, FeedbackSessionResultsBundle > in the existing code itself is redundant. Instead, we should have just used List< FeedbackSessionResultsBundle >, since the key of the map is the feedbackSessionName which can also be can also be obtained from a FeedbackSessionResponseBundle(bundle.feedbackSession.feedbackSessionName).

Using a List, in my opinion will lead to more clean and efficient code, but will require some more code changes. Your opinion on what should I do?